telegram/ext/_callbackqueryhandler.py

```diff
@@ -97,22 +99,21 @@ class CallbackQueryHandler(Handler[Update, CCT]):
 
             .. versionchanged:: 13.6
                Added support for arbitrary callback data.
-        run_async (:obj:`bool`): Determines whether the callback will run asynchronously.
+        block (:obj:`bool`): Determines whether the return value of the callback should be
+            awaited before processing the next handler in
+            :meth:`telegram.ext.Application.process_update`.
 
     """
 
     __slots__ = ('pattern',)
 
     def __init__(
         self,
-        callback: Callable[[Update, CCT], RT],
+        callback: HandlerCallback[Update, CCT, RT],
         pattern: Union[str, Pattern, type, Callable[[object], Optional[bool]]] = None,
```

Comment:
in `check_update`, we should check if `self.pattern` is not a coroutine and if it is, warn.

Reply:
good idea! though I'd rather directly raise a ValueError in `__init__` :)